index.js

```diff
@@ -135,50 +177,63 @@ function nowServing(line) {
   return `Currently serving ${line.shift()}.`
 }
 
-const line = []
+function takeATicketNumber(line) {
+  const line_number = line.push((line.length) + 1)
+  ticketCounter += 1
+  return `Welcome. You are ticket number ${ticketCounter}`
+}
 
+const line = []
+let ticketCounter = 0
 // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+console.log("I am confuuuuused about this one")
+
+const newLine = []
+const takeATicketNumberFunc = function ticketNumberGeneratorFunc (line) {
+  var lineNumber = line.push((line.length) + 1)
+
+return `Welcome. You are ticket number ${lineNumber} `
```

Comment:
This function should return a function!
At the moment, it's returning a string.
Remember that in JS, we treat functions as objects, so we can return a function definition from a function.
So, the solution would look something like this:
```
function takeATicketNumberFunc(){
// ...
   return function(){
   }
})